Add cropdiversityhpc config

TMAdams opened the pull request:
---
name: New Config
about: A new cluster config
---

Please follow these steps before submitting your PR:

- [x] If your PR is a work in progress, include `[WIP]` in its title
- [x] Your PR targets the `master` branch
- [x] You've included links to relevant issues, if any

Steps for adding a new config profile:

- [x] Add your custom config file to the `conf/` directory
- [x] Add your documentation file to the `docs/` directory
- [x] Add your custom profile to the `nfcore_custom.config` file in the top-level directory
- [x] Add your custom profile to the `README.md` file in the top-level directory
- [x] Add your profile name to the `profile:` scope in `.github/workflows/main.yml`
- [x] OPTIONAL: Add your custom profile path and GitHub user name to `.github/CODEOWNERS` (`**/<custom-profile>** @<github-username>`)

<!--
If you require/still waiting for a review, please feel free to request a review from @nf-core/maintainers

Please see uploading to`nf-core/configs` for more details:
https://github.com/nf-core/configs#uploading-to-nf-coreconfigs

Thank you for contributing to nf-core!
-->

maxulysse commented on conf/cropdiversityhpc.config:
```diff
@@ -0,0 +1,20 @@
+    queue = { task.memory <= 32.GB ? (task.time < 6.h ? task.memory <=2.GB ? 'hicpu': 'short' : task.time < 24.h ? 'medium' : 'long') : 'himem' }
+}
+
+conda.enabled = true
+conda.createTimeout = '2 h'
+singularity.enabled = true
+report.overwrite = true
+singularity.pullTimeout = '2 h'
```
conda and singularity both enabled together?
How can user chose?

TMAdams replied:
Ah I see, we'd been testing this on custom workflows we've developed outside of nf-core where a process only had either singularity or conda (if for some reason singularity failed). So having both enabled made sense.

Will do another pass of the docs to see how best to handle this

TMAdams replied:
I'm minded to remove the conda enable option, I've put out a call to our users to see if that will break anyone's workflows and will remove this after I get feedback.

jfy133 replied:
I plus one only supporting the container option. 

And anyway if a user wanted to override singularity, they can just add `-profile <institution>,conda` (obviously won't worked on mixed pipelines but still)

TMAdams replied:
Agreed, I've updated as such

jfy133 commented on conf/cropdiversityhpc.config:
```diff
@@ -0,0 +1,20 @@
+    queue = { task.memory <= 32.GB ? (task.time < 6.h ? task.memory <=2.GB ? 'hicpu': 'short' : task.time < 24.h ? 'medium' : 'long') : 'himem' }
+}
+
+conda.enabled = true
```
Minor niggle, please use the bracket syntax for consistency

TMAdams replied:
Added a commit to resolve this

jfy133 reviewed (approved):
Minor comments but otherwise looks good to me

TMAdams commented:
Can't respond to @jfy133 as I've changed the syntax now, but I've added the override of Singularity option to the docs and rephrased it to heavily encourage conda.

There might be an edge case of mixed pipelines, but we can deal with that on a case by case basis

TMAdams commented:
Corrected indentation width as this was flagged as an error in the automated workflows.

jfy133 commented:
@nf-core-bot fix linting
